### PR TITLE
Do HEAD request first, before polling data for alerts and checks

### PR DESCRIPTION
**Pull request by Jan-M**

### CODE REVIEW

File: src/main/java/de/zalando/zmon/scheduler/ng/BaseSource.java [hunk 1]
```diff
@@ -14,4 +20,20 @@ public BaseSource(String n) {
     public String getName() {
         return name;
     }
+
+    protected static boolean doRefresh(String headerValue, long lastRefresh, Collection lastData) {
+        if(lastData == null) return true;
```
Comment (hjacobs) [inline src/main/java/de/zalando/zmon/scheduler/ng/BaseSource.java]: code format

File: src/main/java/de/zalando/zmon/scheduler/ng/BaseSource.java [hunk 2]
```diff
@@ -14,4 +20,20 @@ public BaseSource(String n) {
     public String getName() {
         return name;
     }
+
+    protected static boolean doRefresh(String headerValue, long lastRefresh, Collection lastData) {
```
Comment (hjacobs) [inline src/main/java/de/zalando/zmon/scheduler/ng/BaseSource.java]: collection is missing generic type "<..>"

File: src/main/java/de/zalando/zmon/scheduler/ng/checks/DefaultCheckSource.java [hunk 3]
```diff
@@ -49,6 +53,15 @@ private HttpHeaders getWithAuth() {
             final String accessToken = tokens.get();
             LOG.info("Querying check definitions with token " + accessToken.substring(0, Math.min(accessToken.length(), 3)) + "..");
             HttpEntity<String> request = new HttpEntity<>(getWithAuth());
+
+            HttpHeaders headers = restTemplate.headForHeaders(url, request);
+            if (headers.containsKey("Last-Modified")) {
+                if (!doRefresh(headers.get("Last-Modified").get(0), lastResultMaxLastModified,lastResults)) {
```
Comment (hjacobs) [inline src/main/java/de/zalando/zmon/scheduler/ng/checks/DefaultCheckSource.java]: missing space after comma

**Comment by hjacobs** [comment]
:+1:

**Comment by Jan-M** [comment]
:+1:

**Pull request merged**